scss/_variables-dark.scss

```diff
@@ -0,0 +1,26 @@
+// Dark color mode variables
+//
+// Custom variables for the `[data-bs-theme="dark"]` theme. Use this as a starting point for your own custom color modes by creating a new theme-specific file like `_variables-dark.scss` and adding the variables you need.
+
+$body-color-dark:                   $gray-500 !default;
```

Comment:
Thinking out loud here.

Here, all the Sass vars use a `-dark` suffix for the dark mode.
For existing components that have a dark variant, we also use the word "dark" in `$navbar-dark-*` and `$dropdown-dark-*` for example.

Not saying this is bad, but we have to think about it in this PR in terms of consistency and so that everything remains understandable.

Something possible would be to use the `-dark` suffix for everything relative to dark mode (to allow the switch thanks to the CSS vars) and rather than having `$navbar-dark-*` and `$dropdown-dark-*` used for the dark variants we could use a word not related to dark/light; something like `$navbar-inverted-*` and `$dropdown-inverted-*`.

|  **Light mode** | **Dark mode**  |
|---|---|
|  $component-bg-color (white) |  $component-bg-color-dark (black or grayish) |
|  $component-inverted-bg-color (black)  |  $component-inverted-color-dark (black or grayish) |

Reply:
I think all *-dark development is not necessary. Just override css variables  in one dark.scss file.

Reply:
Of course, this is not easy. The old one

  - .close-white
  - .navbar-dark
  - .carousel-dark
  - .dropdown-menu-dark

css classes could possibly be moved to bootstrap-compatibility.scss and follow the new [Color schemes pattern](https://github.com/twbs/bootstrap/commit/143f94708335cb706127886db4a8d785046d1bbc#diff-78f864ea896a6a26657a3a38ccb48e5bfda33c9d04e07301a720aa08a3a67ef8L388) everywhere: 

data-bs-theme="light"

``` html
<nav class="navbar navbar-expand-lg" style="background-color: #e3f2fd;" data-bs-theme="light">
  ...
</nav>
```